homeassistant/components/tcp/binary_sensor.py

```diff
@@ -1,22 +1,33 @@
 """Provides a binary sensor which gets its values from a TCP socket."""
+from __future__ import annotations
+
+from typing import Final
+
 from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from .sensor import CONF_VALUE_ON, PLATFORM_SCHEMA, TcpSensor
+from .const import CONF_VALUE_ON
+from .sensor import PLATFORM_SCHEMA as TCP_PLATFORM_SCHEMA, TcpSensor
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({})
+PLATFORM_SCHEMA: Final = TCP_PLATFORM_SCHEMA
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
```

Comment:
It turns out people don't like this alias, let's clean it up.
```suggestion
    discovery_info: dict[str, Any] | None = None,
```

Reply:
mhhh :thinking: I have used it in all my last "typing" PRs :confused:

Reply:
Yeah, me too. Let's keep them as is but not add the new ones.